ui/src/App.tsx

```diff
@@ -64,34 +64,50 @@ function App() {
-            height: '600px',
-            border: '3px solid black',
-            borderRadius: theme.shape.borderRadius,
-            overflow: 'hidden',
-            '& > button': {
-              backgroundColor: 'grey',
-              borderRadius: "0px",
-              '&:not(:nth-child(3n))': {
-                borderRight: 'thin solid black',
-              },
-              '&:nth-child(-n+6)': {
-                borderBottom: 'thin solid black',
-              },
-            },
-          })}
-        >
-          {[...Array(9).keys()].map(entry => <GridButton key={entry}/>)}
-        </Box>
-      </Container>
+        <Container sx={{ py: 2, flex: 1}}>
+          <Typography variant="h4" sx = {{textAlign: 'center'}}>Hello</Typography>
+          <Typography variant="body1" sx = {{textAlign: 'center'}}>Explanation stuff</Typography>
+          <Stack alignItems='center'>
+            <Stack direction='row' alignItems={'center'}>
+              <Typography variant="body1" sx={{textAlign:'center', width:'200px'}}>A category</Typography>
+              <Typography variant="body1" sx={{textAlign:'center', width:'200px'}}>Some other category</Typography>
+              <Typography variant="body1" sx={{textAlign:'center', width:'200px'}}>Looooooooong looooooong multi-line category</Typography>
+            </Stack>
+            <Stack direction='row'>
+              <Stack width='200px' alignItems='center'>
+                <Typography height='200px' variant="body1" sx={{textAlign:'center', alignItems:'center', display:'flex'}}>Author</Typography>
+                <Typography height='200px' variant="body1" sx={{textAlign:'center', alignItems:'center', display:'flex'}}>Some other author with a super-duper multi-line name</Typography>
+                <Typography height='200px' variant="body1" sx={{textAlign:'center', alignItems:'center', display:'flex'}}>One more author</Typography>
+              </Stack>
+              <Box
+                sx={theme => ({
+                  display: 'grid',
+                  gridTemplateColumns: '1fr 1fr 1fr',
+                  gridAutoRows: '1fr',
+                  width: '600px',
+                  height: '600px',
+                  border: '3px solid black',
+                  borderRadius: theme.shape.borderRadius,
+                  overflow: 'hidden',
+                  '& > button': {
+                    backgroundColor: 'grey',
+                    borderRadius: "0px",
+                    '&:not(:nth-child(3n))': {
+                      borderRight: 'thin solid black',
+                    },
+                    '&:nth-child(-n+6)': {
+                      borderBottom: 'thin solid black',
+                    },
+                  },
+                })}
+              >
+                {[...Array(9).keys()].map(entry => <GridButton key={entry}/>)}
+              </Box>
+              {/* Dummy box to take up space: */}
+              <Box height='600px' width='200px'/> 
```

Comment:
que?

Reply:
I wanted the grid itself to be centered along the horizontal axis of the page, so I used the `alignItems='center'` prop for the `Stack` on line 70. However, to the left of the grid is the 200px wide column containing the author names (lines 76-81), so the grid became slightly off-centre. I therefore added this 200px wide "dummy box" on line 107 to the _right_ of the grid, in order to ensure that the grid is in the centre of the page.

I realize it's a bit hack-y. I'd like to have some way to say "centre the grid, and then put the authors column immediately to the left of that." I had hoped to do this using absolute/relative positioning, but I wasn't able to find a solution.